catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -230,32 +207,16 @@ public boolean sourceStatus(String servicePID) {
                             source.put(MAP_ENTRY_NAME, config.getPid());
                         }
 
-                        Dictionary<String, Object> configProperties = config.getProperties();
-                        List<Map<String, Object>> metatypeProperties =
-                                (List<Map<String, Object>>) metatype.get("metatype");
+                        Dictionary<String, Object> properties = config.getProperties();
                         Map<String, Object> plist = new HashMap<>();
-                        Object propertyValue;
-
-                        for (String key : Collections.list(configProperties.keys())) {
-                            propertyValue = configProperties.get(key);
-                            // If the configuration property is a password, set its value to
-                            // "password" if it's not empty, so that the real password value will
-                            // be hidden. (Passwords within source configs are usually optional,
-                            // thus it should have the ability to update to an empty string).
-                            if (!"".equals(propertyValue)) {
-                                for (Map<String, Object> metatypeProperty : metatypeProperties) {
-                                    if (metatypeProperty.get("id")
-                                            .equals(key) && AttributeDefinition.PASSWORD
-                                            == (Integer) metatypeProperty.get("type")) {
-                                        propertyValue = "password";
-                                        break;
-                                    }
-                                }
-                            }
-                            plist.put(key, propertyValue);
+                        for (String key : Collections.list(properties.keys())) {
+                            plist.put(key, properties.get(key));
                         }
                         source.put(MAP_ENTRY_PROPERTIES, plist);
-
+                        source.put(MAP_ENTRY_REPORT_ACTIONS,
+                                getActions(config, reportActionProviders));
+                        source.put(MAP_ENTRY_OPERATION_ACTIONS,
+                                getActions(config, operationActionProviders));
                         configurations.add(source);
```

Comment:
Just curious... `configurations` is a `List<Map<String, Object>>` that contains a `source` which is itself a `Map<String, Object>`, which contains the `MAP_ENTRY_OPERATION_ACTIONS` entry whose value is of type `List<Map<String, String>>`, right? My head hurts :-(